### PR TITLE
Adding unmerged test cases for APIProviderImpl 

### LINKED ISSUE #4769: Adding unmerged unit test cases for APIProviderImpl
**Description:**
This PR adds/merges the APIProviderImpl test cases which were added to another branch (by @lakmali  and @krish), but not committed to 6.x branch.

### PR DESCRIPTION
**Pull request by rmsamitha**
### Proposed changes in this pull request

- This PR adds/merges the APIProviderImpl test cases which were added to another branch (by @lakmali  and @krish), but not committed to 6.x branch.
- Related changes are done also to APIProviderImpl too.

[List all changes you want to add here. If you fixed an issue, please
add a reference to that issue as well.]
- This Fixes #4769

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 1]
```diff
@@ -3316,7 +3334,9 @@ private boolean addExternalAPIStoresDetails(APIIdentifier apiId,Set<APIStore> ap
                 }
             }
 		} catch (Exception e) {
-			handleException("Issue is in getting custom InSequences from the Registry", e);
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Why is this removed?

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 2]
```diff
@@ -3316,7 +3334,9 @@ private boolean addExternalAPIStoresDetails(APIIdentifier apiId,Set<APIStore> ap
...
+            String msg = "Issue is in getting custom InSequences from the Registry";
+            log.error(msg, e);
+            throw new APIManagementException(msg, e);
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: formatting error

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 3]
```diff
@@ -2839,7 +2845,7 @@ public void deleteAPI(APIIdentifier identifier) throws APIManagementException {
                 foundApiList = searchAPIs(searchTerm, searchType);
 			}
 		} catch (APIManagementException e) {
-			handleException("Failed to search APIs with type", e);
+            handleException("Failed to search APIs with type", e);
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: formatting error

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java [hunk 4]
```diff
@@ -2925,8 +2931,8 @@ public void deleteAPI(APIIdentifier identifier) throws APIManagementException {
 
 			}
 		} catch (RegistryException e) {
-			handleException("Failed to search APIs with type", e);
-		} finally {
+            handleException("Failed to search APIs with type", e);
+        } finally {
```
Comment (bhathiya) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: formatting error

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java (hunk 1 above)
Comment (rmsamitha) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java]: Added handleException() back
